src/browser/api/system.js

```diff
@@ -248,7 +248,20 @@ exports.System = {
         return hash.digest('hex');
     },
     getDeviceId: function() {
-        return electronApp.getHostToken();
+        if (process.platform === 'win32') {
+            return electronApp.getHostToken();
+        } else {
+            const hash = crypto.createHash('sha256');
+
+            let macAddress = os.networkInterfaces().en0[0].mac;
```

Comment:
safeguard os.networkInterfaces().en0 and throw exception ?

Reply:
Done